src/main/scala/calc/NodeCompiler.scala

```diff
@@ -0,0 +1,31 @@
+package calc
+
+import org.scalajs.core.ir.{Trees => irt}
+
+object NodeCompiler {
+  def apply(f: Tree => irt.Tree): NodeCompiler = {
```

Comment:
That is the first time I ever see anything like that. It's going to be crazy inefficient on any non-trivial programs.

At least, instead of catching `CompileError`s, you should use a `PartialFunction` and chain things with `orElse`. But even that is a very unorthodox way of writing a compiling. Why did you not use a simple pattern matching? More efficient and more concise, what is there not to love?